mypy/mixedtraverser.py

```diff
@@ -0,0 +1,89 @@
+from typing import Optional
+
+from mypy.nodes import (
+    MypyFile, Var, FuncItem, ClassDef, AssignmentStmt, OperatorAssignmentStmt, ForStmt, WithStmt,
+    CastExpr, TypeApplication, TypeAliasExpr, TypeVarExpr, TypedDictExpr, NamedTupleExpr,
+    EnumCallExpr, PromoteExpr, NewTypeExpr
+)
+from mypy.types import Type
+from mypy.traverser import TraverserVisitor
+from mypy.typetraverser import TypeTraverserVisitor
+
+
+class MixedTraverserVisitor(TraverserVisitor, TypeTraverserVisitor):
+    """Recursive traversal of both Node and Type objects."""
+
+    # Symbol nodes
+
+    def visit_var(self, var: Var) -> None:
+        self.visit_optional_type(var.type)
+
+    def visit_func(self, o: FuncItem) -> None:
+        super().visit_func(o)
+        self.visit_optional_type(o.type)
+
+    def visit_class_def(self, o: ClassDef) -> None:
+        super().visit_class_def(o)
+        info = o.info
+        if info:
+            for base in info.bases:
+                base.accept(self)
+            self.visit_optional_type(info.typeddict_type)
```

Comment:
Why this should also be here? This one should be reached as `o.analyzed.info.typeddict_type` from `visit_typeddict_expr()`.

Also if there is any reason for typed dict, why then not named tuple is not here?

Reply:
This looks redundant. I'll remove this.